File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {

    public boolean isYelling(String input) {
        boolean hasLetters = input.chars().filter(i -> Character.isLetter((char) i)).sum() > 0;
        boolean allLettersAreUppers = input.chars().filter(i -> Character.isLetter((char) i))
                .allMatch(i -> Character.isUpperCase((char) i));
        return hasLetters && allLettersAreUppers;
    }

    public boolean isQuestion(String input) {
        return input.length() > 0 && input.toCharArray()[input.length() - 1] == '?';
    }

    public String hey(String input) {
        input = input.strip();
        if (input == "")
            return "Fine. Be that way!";

        if (isQuestion(input)) {
            return isYelling(input) ? "Calm down, I know what I'm doing!" : "Sure.";
        }

        return isYelling(input) ? "Whoa, chill out!" : "Whatever.";
    }
}

File: solutions/python/bob/1/bob.py

def response(hey_bob: str) -> str:
    hey_bob = hey_bob.strip()
    is_silent = all(s.isspace() for s in hey_bob)
    chars = [s for s in hey_bob if s.isalpha()]
    is_yelling = len(chars) > 0 and all(s.isupper() for s in chars)
    is_question = hey_bob.endswith("?")

    if not is_yelling and is_question:
        return "Sure."

    if is_yelling and not is_question:
        return "Whoa, chill out!"

    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"

    if is_silent:
        return "Fine. Be that way!"

    return "Whatever."